R/modelling.R

```diff
@@ -172,24 +173,12 @@ fit_seromodel <- function(serodata,
   )
 
   if (class(seromodel_fit@sim$samples) != "NULL") {
```

Comment:
In theory, you're not supposed to `@` directly on a S4 object like `seromodel_fit()`, but use accessor functions instead: https://mc-stan.org/rstan/articles/stanfit_objects.html.

However, it's a common pattern, even if not part of the official spec so if it's too much of a struggle to use accessors, you can probably keep using `@`.

Additional discussion on this topic: https://stackoverflow.com/questions/9900134/is-it-bad-practice-to-access-s4-objects-slots-directly-using

Reply:
I'm guessing that this alternative:
```
 sim <- attr(seromodel_fit, "sim")
 if (class(sim$samples) != "NULL")
```
suffers similar potential drawbacks than `@`. If not, and you think is a better alternative I'll change it. Otherwise I'll leave it as it is for the time being.

Reply:
Yes, you're right that rstan doesn't provide accessor functions for all the information stored in the stanfit object so let's stay with `@`.

Reply:
The last bullet in the [motivation for S7](https://rconsortium.github.io/S7/articles/motivation.html) actually talks about the confusion about using @ for accessing object internals.